src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -424,23 +395,11 @@ public sealed override bool Equals(Symbol obj, TypeCompareKind compareKind)
 
         public override int GetHashCode()
         {
-            int code = _hashCode;
-
-            if (code == 0)
-            {
-                code = ComputeHashCode();
-
-                // 0 means that hashcode is not initialized. 
-                // in a case we really get 0 for the hashcode, tweak it by +1
-                if (code == 0)
-                {
-                    code++;
-                }
-
-                _hashCode = code;
-            }
-
-            return code;
+            // Substituted methods can still be considered equal to other method symbols when 
+            // the substituted type is considered equal (e.g. when ignoring nullability).
+            // We defer the hashcode to the original definition so it will match in those scenarios. 
+            // See https://github.com/dotnet/roslyn/issues/38195
+            return this.OriginalDefinition.GetHashCode();
```

Comment:
>return this.OriginalDefinition.GetHashCode(); [](start = 12, length = 45)

Same comment as for substituted fields, we could keep original calculation under some conditions.